www/js/app.js

```diff
@@ -98,6 +110,20 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     
     // Unique identifier of the article expected to be displayed
     var expectedArticleURLToBeDisplayed = "";
+
+    //handles if auto Theme is selected
+    function handleAutoGenTheme() { 
+        if (window.matchMedia && darkPreference.matches) uiUtil.applyAppTheme('dark_invert');
+        else uiUtil.applyAppTheme('light');
+        // darkPreference.addEventListener('change', function (e) {
+        //     var theme = e.matches ? 'dark_invert' : 'light'
+        //     uiUtil.applyAppTheme(theme);
+        //   })
+     }
+     function handleAutoWikiTheme() { 
+        if (window.matchMedia && darkPreference.matches) uiUtil.applyAppTheme('dark_mwInvert');
+        else uiUtil.applyAppTheme('light');
+     }
```

Comment:
As mentioned above, don't have separate handlers for each function. Handle whatever value is set in `uiUtil.applyAppTheme()`.

Reply:
Resolved